index.php

```diff
@@ -94,10 +94,34 @@
         ));
     }
 
+    global $CFG;
+    if (!empty($CFG->rememberusername)) {
+        $cookiename = 'REMEMBERUNAME';
+        if (isset($_COOKIE[$cookiename])) {
+            $username = rc4decrypt($_COOKIE[$cookiename]);
+        }
+        global $frm;
+        $frm->username = $username;
+        $passwordform->set_data(array(
+            'username' => $username,
+            'rememberme' => 1,
+        ));
+    }
+
     if ($passwordform->is_submitted() &&
         $passwordform->is_validated()) {
         $values = $passwordform->get_data();
         $helper->set_passform_params_in_auth_global_vars($values);
+
+        //Setting loginas cookie
+        if (!empty($values->rememberme)) {
+            global $CFG;
+            // Set username cookie for 60 days.
+            $cookiesecure = is_moodle_cookie_secure();
+            $Loggedinusername = $values->username;
+            setcookie('REMEMBERUNAME', rc4encrypt($Loggedinusername), time() + (DAYSECS * 60), $CFG->sessioncookiepath,
```

Comment:
Not compulsory, just to make it more readable can we break the args in multiple lines